src/main/java/org/folio/dew/service/BulkEditParseService.java

```diff
@@ -0,0 +1,184 @@
+    user.setUpdatedDate(getDate(userFormat.getUpdatedDate()));
+    user.setTags(getTags(userFormat));
+    user.setCustomFields(getCustomFields(userFormat));
+  }
+
+  private boolean getIsActive(UserFormat userFormat) {
+    String value = userFormat.getActive();
+    if (value.matches("true") || value.matches("false")) {
+      return Boolean.parseBoolean(value);
+    }
+    //TODO in MODBULKED-14 save error that filed has a wrong value instead of returning false
+    return false;
+  }
+
+  private String getPatronGroupId(UserFormat userFormat) {
+    if (isNotEmpty(userFormat.getPatronGroup())) {
+      UserGroupCollection userGroup = userReferenceService.getUserGroupByGroupName(userFormat.getPatronGroup());
+      return userGroup.getUsergroups().iterator().next().getId();
+    }
+    return null;
+  }
+
+  private List<String> getUserDepartments(UserFormat userFormat) {
+    String[] departmentNames = userFormat.getDepartments().split(ARRAY_DELIMITER);
+    return Arrays.stream(departmentNames).parallel()
+      .map(userReferenceService::getDepartmentByName)
+      .flatMap(departmentCollection -> departmentCollection.getDepartments().stream())
+      .map(Department::getId)
+      .collect(Collectors.toList());
+  }
+
+  private List<String> getProxyFor(UserFormat userFormat) {
+    String[] proxyUserNames = userFormat.getProxyFor().split(ARRAY_DELIMITER);
+    return Arrays.stream(proxyUserNames)
+      .parallel()
+      .map(userReferenceService::getUserByName)
+      .flatMap(userCollection -> userCollection.getUsers().stream())
+      .map(User::getId)
+      .map(userReferenceService::getProxyForByProxyUserId)
+      .flatMap(proxyForCollection -> proxyForCollection.getProxiesFor().stream())
+      .map(ProxyFor::getId)
+      .collect(Collectors.toList());
+  }
+
+  private Personal getUserPersonalInfo(UserFormat userFormat) {
+    Personal personal = new Personal();
+    personal.setLastName(userFormat.getLastName());
+    personal.setFirstName(userFormat.getFirstName());
+    personal.setMiddleName(userFormat.getMiddleName());
+    personal.setPreferredFirstName(userFormat.getPreferredFirstName());
+    personal.setEmail(userFormat.getEmail());
+    personal.setPhone(userFormat.getPhone());
+    personal.setMobilePhone(userFormat.getMobilePhone());
+    personal.setDateOfBirth(getDate(userFormat.getDateOfBirth()));
+    personal.setAddresses(getUserAddresses(userFormat));
+    personal.setPreferredContactTypeId(userFormat.getPreferredContactTypeId());
+    return personal;
+  }
+
+  private Date getDate(String date) {
+    LocalDateTime localDateTime = LocalDateTime.parse(date, DateTimeFormatter.ofPattern(DATE_TIME_PATTERN));
+    return Date.from(localDateTime.atZone(UTC).toInstant());
+  }
+
+  private List<Address> getUserAddresses(UserFormat userFormat) {
+    String[] addresses = userFormat.getAddresses().split(ITEM_DELIMITER_PATTERN);
+    return Arrays.stream(addresses)
```

Comment:
Empty strings should be excluded from processing

Reply:
Corrected